sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/Shared/MqttConnectEventErrorResponseDeserializationHelper.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebPubSub.Common;
+
+#nullable enable
+
+internal class MqttConnectEventErrorResponseDeserializationHelper : MqttConnectEventErrorResponse
+{
+    public MqttConnectEventErrorResponseDeserializationHelper(MqttConnectEventErrorResponsePropertiesDeserializationHelper mqtt) : base(new MqttConnectEventErrorResponseProperties((MqttV500ConnectReasonCode)mqtt.Code))
+    {
+        Mqtt.Reason = mqtt.Reason;
```

Comment:
Now we don't allow passing an integer to the MqttConnectEventErrorResponseProperties ctor to initialize the `code` field, to avoid an invalid `code`. Therefore, we have to use a tricky way to convert the integer code to a strongly typed `MqttV500ConnectReasonCode` or `MqttV311ConnectReturnCode`. This conversion doesn't lose information. If the isolated worker passes an invalid code, we won't check it in the SDK side, leaving it to the server to check, which is a general practice in Azure SDK to avoid client side verification.